[obrona] iP

obrona opened the pull request:
# Duch, the best chat bot 😄 🐞 
> Organise your life.
> Never forget anything else again.
+ **text based**
+ **easy to lean**
+ ***fast and lightweight***

## it is totally free

## All you need to do is
1. Download the jar file [here](https://github.com/obrona/ip/releases/tag/A-Jar)
2. Download java 17 [here](https://www.oracle.com/sg/java/technologies/downloads/#java17)
3. install java 
4. type `java -jar Duch.jar` on your command line

## Features
- [x] Manage deadline
- [x] Manage event
- [x] Manage todo
- [ ] Sort events (to be added)

## Examples of java code used in this project
```java
package Duch;

import java.util.ArrayList;
import Duch.Task.Task;

/**
 * Stores a list of Tasks
 */
public class TaskList extends ArrayList<Task> {
    public TaskList() {
        super();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Level-0

Commit:
Level-1

Commit:
Level-2

Commit:
Level-3

Commit:
Level-4

Commit:
A-TextUiTesting

Commit:
Level-5

Commit:
Level-6

Commit:
Level-7

Commit:
Merge branch 'branch-Level-7'

Commit:
Level-8

Commit:
Merge branch 'branch-Level-8'

Commit:
A-MoreOOP: Use more OOP

Commit:
A-Packages: Organize into packages

Commit:
Merge branch 'add-gradle-support' of https://github.com/obrona/ip into add-gradle-support

Commit:
A-Gradle: Use Gradle

Commit:
A-JUnit: Add JUnit Tests

Commit:
A-Jar: Create a JAR File

Commit:
A-JavaDoc: Add JavaDoc

Commit:
A-CodingStandard: Follow the Coding Standard

Commit:
Level-9: Add Find

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'branch-Level-9'

Commit:
Level 10: Add GUI

Commit:
Level 10: Add .vscode to gitignore

Commit:
Merge branch 'branch-Level-10'

jiaaaaen commented on src/main/java/Duch/Parser.java:
```diff
@@ -0,0 +1,218 @@
+        return ui.print("OK, I've marked this task as not done yet:\n  " + t.toString() + "\n");
+    }
+
+    private String printDelete(Task t) {
```
u can include these print methods in the ui class!

jiaaaaen commented on src/main/java/Duch/Parser.java:
```diff
@@ -0,0 +1,218 @@
+     * 
+     * @param arr The splitted command
+     */
+    public String unmark(String[] arr) {
```
method name shld be unMark instead

jiaaaaen reviewed (commented):
looks good! very organised and easy to understand :)

jiaaaaen commented on src/main/java/Duch/Parser.java:
```diff
@@ -0,0 +1,218 @@
+     * @return
+     * @throws DuchException
+     */
+    public String handleCommand(String cmd) throws DuchException {
```
very neat handling!

jiaaaaen commented on src/main/java/Duch/Storage.java:
```diff
@@ -0,0 +1,111 @@
+     * @param str The stored deadline string
+     * 
+     */
+    public void parseDeadline(String str) {
```
parsing could be done in the parser class instead!

jiaaaaen reviewed (commented):
looks good! very organised and easy to understand :)

Commit:
Add CSS

t-leongchuan commented on src/main/java/Duch/Storage.java:
```diff
@@ -0,0 +1,111 @@
+    /**
+     * Open the ./data/text file, where the tasks will be stored
+     */
+    public void openFile() {
+        Path path = Paths.get(pathString);
+        Path directory = path.getParent();
+        try {
+            if (Files.notExists(directory)) {
+                Files.createDirectories(directory);
+            }
+
+            if (Files.notExists(path)) {
+                Files.createFile(path);
+            }
+        } catch (IOException e) {System.out.println(e.getMessage());}
+        
+        // start reading from file
+        try {
+            Scanner scanner = new Scanner(Files.newBufferedReader(path));
+            while (scanner.hasNextLine()) {
+                String str = scanner.nextLine();
+                if (str.length() == 0) {
+                    break;
+                }
+                
+                char type = str.charAt(0);
+                if (type == 'T') {
+                    parseTodo(str);
+                } else if (type == 'E') {
+                    parseEvent(str);
+                } else if (type == 'D') {
+                    parseDeadline(str);
+                }
+            }
+            scanner.close();
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        } 
+    }
```
Code Quality (Naming) LGTM. Naming is well structured and easy to read. Well Done! 👍

t-leongchuan commented on src/main/java/Duch/Ui.java:
```diff
@@ -0,0 +1,17 @@
+ * Handles formatting text that needs to be printed on stdout
+ */
+public class Ui {
+    String space = "    ";
```
Should this be declared constant instead?

Commit:
A-FullCommitMessage: write a full commit message

We have not written a ful commit message

Writing a full commit message will help us to be better
software engineers

Let's write a full commit message

Commit:
A-Assertions: Use Assertions

Commit:
A-CodeQuality: Improve code quality

Some of the files have poor code quality

Let's improve code quality by doing some refactoring

Commit:
A-CodeQuality: Improve code quality

Some of the files have poor code quality

Let's improve code quality by doing some refactoring

obrona committed:
Merge pull request #1 from obrona/branch-A-Assertions

A-Assertions: Use Assertions

Commit:
BCD-Extension: Add an extension

Commit:
BCD-Extension: Change Parser undo() message

obrona committed:
A-CI: Create main.yml

Commit:
Untrack .vscode

obrona committed:
Merge pull request #3 from obrona/branch-A-CodeQuality

A-CodeQuality: Improve code quality

Commit:
Edit javadocs comments

Commit:
Update package names and README

Commit:
Change README title

Commit:
Update javadoc comments

Commit:
Prevent adding dialog box  when user input is empty

Commit:
Add more tests

Commit:
Update README.md

Commit:
Edit ReadMe

Commit:
Edit Readme
